### PR TITLE
fix: return all info from AuthInfo.listAllAuthorizations

### PR DESCRIPTION
**Pull request by mdonnalley**
Return more information from `AuthInfo.listAllAuthorizations`

@W-9670252@
@ W-9680192@

### CODE REVIEW

**Review by peternhale — CHANGES_REQUESTED** [review]
@mdonnalley I was expecting a change to SfOrg
```
export type SfOrg = {
  alias: Optional<string>;
  username: Optional<string>;
  orgId: Optional<string>;
  instanceUrl: Optional<string>;
  accessToken?: string;
  oauthMethod?: 'jwt' | 'web' | 'token' | 'unknown';
  error?: string;
} & SfEntry;
```

to 

```export type SfOrg = {
  aliases: Optional<string[]>;
  username: Optional<string>;
  orgId: Optional<string>;
  instanceUrl: Optional<string>;
  accessToken?: string;
  oauthMethod?: 'jwt' | 'web' | 'token' | 'unknown';
  error?: string;
} & SfEntry;
```

**Review by peternhale — APPROVED** [review]

**Review by peternhale — APPROVED** [review]

File: src/org/authInfo.ts [hunk 1]
```diff
@@ -373,40 +385,47 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
...
+        const { orgId, instanceUrl, devHubUsername } = authInfo.getFields();
+        const isScratchOrg = Boolean(devHubUsername);
         final.push({
-          alias,
+          aliases,
+          configs,
           username,
-          orgId,
           instanceUrl,
+          isScratchOrg,
```
Comment (RodEsp) [inline src/org/authInfo.ts:410]:
```suggestion
          isScratchOrg = Boolean(devHubUsername),
```

File: src/org/authInfo.ts [hunk 2]
```diff
@@ -373,40 +385,47 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
...
-      const alias = globalInfo.aliases.get(username) ?? undefined;
+      const aliases = globalInfo.aliases.getAll(username) ?? undefined;
+      const configs = config
+        .filter((c) => aliases.includes(c.value as string) || c.value === username)
+        .map((c) => c.key);
       try {
         const authInfo = await AuthInfo.create({ username });
-        const { orgId, instanceUrl } = authInfo.getFields();
+        const { orgId, instanceUrl, devHubUsername } = authInfo.getFields();
+        const isScratchOrg = Boolean(devHubUsername);
```
Comment (RodEsp) [inline src/org/authInfo.ts:404]:
```suggestion
        const { orgId, instanceUrl, devHubUsername } = authInfo.getFields();
```

File: src/org/authInfo.ts [hunk 3]
```diff
@@ -373,40 +385,47 @@ export class AuthInfo extends AsyncOptionalCreatable<AuthInfo.Options> {
...
     const auths = Object.values(globalInfo.orgs.getAll());
-    const final: SfOrg[] = [];
+    const final: OrgAuthorization[] = [];
     for (const auth of auths) {
       const username = ensureString(auth.username);
-      const alias = globalInfo.aliases.get(username) ?? undefined;
+      const aliases = globalInfo.aliases.getAll(username) ?? undefined;
+      const configs = config
+        .filter((c) => aliases.includes(c.value as string) || c.value === username)
+        .map((c) => c.key);
```
Comment (RodEsp) [inline src/org/authInfo.ts:400]: Can we put a comment here about what exactly this is supposed to be doing please?

**Review by RodEsp — CHANGES_REQUESTED** [review]

**Review by RodEsp — APPROVED** [review]

**Pull request merged**